components/eam/src/physics/crm/samomp/crm_module.F90

```diff
@@ -0,0 +1,2499 @@
+
+module crm_module
+  use perf_mod
+  use task_init_mod, only: task_init
+  use params_kind, only: crm_rknd, r8
+  use abcoefs_mod, only: abcoefs
+  use kurant_mod, only: kurant
+  use setperturb_mod, only: setperturb
+  use boundaries_mod, only: boundaries
+  use forcing_mod, only: forcing
+  use advect_mom_mod, only: advect_mom
+  use adams_mod, only: adams
+  use advect_all_scalars_mod, only: advect_all_scalars
+  use sat_mod
+  use crm_surface_mod
+#ifdef sam1mom
+  use precip_init_mod
+#endif
+  use zero_mod
+  use buoyancy_mod
+  use pressure_mod
+  use uvw_mod
+  use diagnose_mod
+  use damping_mod
+  use ice_fall_mod
+  use coriolis_mod
+
+  use crm_state_module,       only: crm_state_type
+  use crm_rad_module,         only: crm_rad_type
+  use crm_input_module,       only: crm_input_type
+  use crm_output_module,      only: crm_output_type
+  use crm_ecpp_output_module, only: crm_ecpp_output_type
+#ifdef ECPP  
+  use module_ecpp_crm_driver, only: ecpp_crm_stat, ecpp_crm_init, ecpp_crm_cleanup
+#endif
+!---------------------------------------------------------------
+!  Super-parameterization's main driver
+!  Marat Khairoutdinov, 2001-2009
+!---------------------------------------------------------------
+use setparm_mod, only : setparm
+
+contains
+
+subroutine crm(lchnk, ncrms, dt_gl, plev,       &
+                crm_input, crm_state, crm_rad,  &
+                crm_ecpp_output, crm_output, crm_clear_rh, &
+                latitude0, longitude0, gcolp, igstep, &
+                use_crm_accel_in, crm_accel_factor_in, crm_accel_uv_in)
```

Comment:
Arguments for variance transport have since been added. In order to support the same function call, this interface needs to include these arguments, and then should assert that variance transport is not requested since it is not implemented in this version of the code.